browser/about_flags.cc

```diff
@@ -486,6 +502,33 @@ constexpr char kPlaylistDescription[] = "Enables Playlist";
 #define PLAYLIST_FEATURE_ENTRIES
 #endif
 
+#if defined(TOOLKIT_VIEWS)
+uint16_t AllowForDevVersion(uint16_t os) {
+#if BUILDFLAG(IS_WIN)
+  if (!install_static::InstallDetails::IsSet()) {
+    // During unit test, this can happen.
+    return 0;
+  }
+#endif
+  if (auto channel = chrome::GetChannel();
+      channel == version_info::Channel::STABLE ||
+      channel == version_info::Channel::BETA) {
+    return 0;
+  }
+
+  return os;
+}
+
+#define BRAVE_VERTICAL_TABS_FEATURE_ENTRY \
+    {kBraveVerticalTabsFeatureInternalName,  \
+    flag_descriptions::kBraveVerticalTabsName, \
+    flag_descriptions::kBraveVerticalTabsDescription, \
+    AllowForDevVersion(kOsWin | kOsMac), \
```

Comment:
this is gated on toolkit views everywhere else so we should conditionally include this based on toolkit views, not os. The channel portion will be handled by griffin